### ISSUE TITLE
Unexpected `subgraph()` results in undirected graphs

### ISSUE THREAD

**Issue by merkys** [graphviz-perl/Graph#28]
This may be just me, but reading

> An edge is added to the subgraph if there is an edge in the original graph that starts from the `src` set of vertices and ends in the `dst` set of vertices.

I understand this should also work on undirected graphs. However, the following gives no results:
```perl
my $g = Graph::Undirected->new;
$g->add_edge( "A", "B" );
print $g->subgraph( ["A"], ["B"] )->edges;
```

Are `src` and `dst` supposed to be used only with directed graphs? If so, the following sentence also is a bit confusing

> You can leave out dst in which case dst is assumed to be the same: this is called a *vertex-induced subgraph*.

as `subgraph()` with `src` only works correctly with undirected graphs.

**Edit**: Observed in v0.9725.

**Comment by mohawk2** [comment]
This was caused by not testing `subgraph` with undirected graphs, and me not using the Set::Object `contains` method correctly with those. I have captured this in the `subgraph` tests and am working on a fix.

**Comment by mohawk2** [comment]
By the way, this was released immediately as 0.9726. Please report if it has not in fact fixed this problem!

**Comment by merkys** [comment]
Thanks for prompt fix! I have tested 0.9726, all seems to work fine now.
